src/assets/react/sagas/pdf.js

```diff
@@ -132,6 +138,40 @@ export function * generatePdf ({ payload }) {
   yield delay(1000)
 }
 
+export function * generateDownloadZipUrl (generatePdfList, sessionId) {
+  try {
+    const response = yield retry(3, 1000, apiRequestGeneratePdfZip, sessionId)
+    const responseBody = yield response.json()
+
+    if (!response.ok || !responseBody.downloadUrl) {
+      throw response
+    }
+
+    yield put({ type: GENERATE_DOWNLOAD_ZIP_URL, payload: responseBody.downloadUrl })
+  } catch (error) {
+    // To DO
+  }
+
+  /* Redux state for downloadZipUrl */
+  const downloadZipUrl = yield select(getStateDownloadZipUrl)
+
+  /* If generatePdfList array content are all processed, request download zip */
```

Comment:
I'd like to move this to its own function and then `yield` it  after line 207 in the function below. 

```suggestion
  /* If generatePdfList array content are all processed, request download zip */
   if (generatePdfList.length === 0) {
    try {
      const response = yield call(apiRequestDownloadZipFile, downloadZipUrl)

      if (!response.ok) {
        throw response
      }

      /* Auto download the generated PDF zip file */
      window.location.assign(downloadZipUrl)
    } catch(error) {
      /* todo fatal error */
    }
  }
```